lisa/microsoft/testsuites/storage/dmcache.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from typing import Any
+
+from assertpy.assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CBLMariner, Posix
+from lisa.testsuite import simple_requirement
+from lisa.tools import Mkdir, Mkfs, Modprobe, Mount
+from lisa.tools.dmsetup import Dmsetup
+from lisa.tools.losetup import Losetup
+from lisa.tools.lvconvert import Lvconvert
+from lisa.tools.lvcreate import Lvcreate
+from lisa.tools.lvremove import Lvremove
+from lisa.tools.lvs import Lvs
+from lisa.tools.mkfs import FileSystem
+from lisa.tools.pvcreate import Pvcreate
+from lisa.tools.pvremove import Pvremove
+from lisa.tools.vgcreate import Vgcreate
+from lisa.tools.vgremove import Vgremove
+from lisa.tools.vgs import Vgs
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="dmcache",
+    category="functional",
+    description="""
+    This test suite validates dm-cache functionality using LVM on Azure VMs.
+    It sets up a dm-cache configuration to verify that caching is functional
+    and provides performance benefits.
+    """,
+)
+class DmCacheTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+
+        # Check if this is a POSIX system
+        assert isinstance(node.os, Posix), f"{node.os} is not supported"
+
+        # Try to load dm-cache module if not already loaded
+        modprobe = node.tools[Modprobe]
+        try:
+            modprobe.load("dm-cache")
+        except AssertionError:
+            log.warning("Failed to load dm-cache module")
+            raise SkippedException(
+                "dm-cache module is not available or cannot be loaded"
+            )
+
+        # Check if LVM tools are available
+        # result = node.execute("which pvcreate", no_error_log=True)
+        # if result.exit_code != 0:
+        #     raise SkippedException("LVM tools are not available on this system")
+
+    @TestCaseMetadata(
+        description="""
+        This test verifies dm-cache functionality by:
+        1. Creating loopback devices to simulate slow origin and fast cache disks
+        2. Setting up LVM physical volumes and volume groups
+        3. Creating logical volumes for origin and cache pool
+        4. Attaching cache pool to origin LV to enable caching
+        5. Formatting and mounting the cached logical volume
+        6. Verifying the dm-cache setup is working correctly
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[CBLMariner],
+        ),
+    )
+    def verify_dm_cache_setup(
```

Comment:
How long does the test case take?

Reply:
It was less than 10 seconds on my local machine

`2025-11-06 22:33:41.365[139678544017088][INFO] lisa.case[verify_dm_cache_setup][lisa_0_0] result: PASSED, elapsed: 9.442 sec`